x-pack/plugins/synthetics/public/apps/synthetics/hooks/use_monitors_sorted_by_status.tsx

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useEffect, useMemo, useState, useRef } from 'react';
+import { isEqual } from 'lodash';
+import { useEsSearch } from '@kbn/observability-plugin/public';
+import { useSelector } from 'react-redux';
+import { useParams } from 'react-router-dom';
+import { Ping, MonitorOverviewItem } from '../../../../common/runtime_types';
+import {
+  EXCLUDE_RUN_ONCE_FILTER,
+  getTimeSpanFilter,
+  SUMMARY_FILTER,
+} from '../../../../common/constants/client_defaults';
+import { SYNTHETICS_INDEX_PATTERN, UNNAMED_LOCATION } from '../../../../common/constants';
+import { selectOverviewState } from '../state/overview';
+import { useSyntheticsRefreshContext } from '../contexts';
+import { useLocationNames } from './use_location_names';
+
+export function useMonitorsSortedByStatus(shouldUpdate: boolean) {
+  const {
+    pageState: { sortOrder },
+    data: { allMonitorIds, monitors },
+  } = useSelector(selectOverviewState);
+  const [monitorsSortedByStatus, setMonitorsSortedByStatus] = useState<
+    Record<string, MonitorOverviewItem[]>
+  >({ up: [], down: [], disabled: [] });
+  const downMonitors = useRef<Record<string, string[]> | null>(null);
+  const currentMonitors = useRef<MonitorOverviewItem[] | null>(monitors);
+  const locationNames = useLocationNames();
+  const { lastRefresh } = useSyntheticsRefreshContext();
+
+  const { monitorId } = useParams<{ monitorId: string }>();
```

Comment:
This is a mistake. One id is not relevant in this flow. Should filter by multiple ids.